pynxtools/dataconverter/readers/xps/xy/xy_specs.py

```diff
@@ -0,0 +1,811 @@
+                "dwell_time",
+            ],
+            "region": [
+                "analysis_method",
+                "spectrum_type",
+                "dwell_time",
+                "comments",
+                "spectrum_id",
+                "time_stamp",
+                "scans",
+                "spectrum_id",
+                "cycle_no",
+            ],
+        }
+
+        for spectrum in spectra:
+            self._update_xps_dict_with_spectrum(spectrum, key_map)
+
+    def _update_xps_dict_with_spectrum(self, spectrum, key_map):
+        """
+        Map one spectrum from raw data to NXmpes-ready dict.
+
+        """
+        # pylint: disable=too-many-locals
+        # pylint: disable=duplicate-code
+        group_parent = f'{self._root_path}/RegionGroup_{spectrum["group_name"]}'
+        region_parent = f'{group_parent}/regions/RegionData_{spectrum["region_name"]}'
+        file_parent = f"{region_parent}/file_info"
+        instrument_parent = f"{region_parent}/instrument"
+        analyser_parent = f"{instrument_parent}/analyser"
+
+        path_map = {
+            "file_info": f"{file_parent}",
+            "user": f"{region_parent}/user",
+            "instrument": f"{instrument_parent}",
+            "source": f"{instrument_parent}/source",
+            "beam": f"{instrument_parent}/beam",
+            "analyser": f"{analyser_parent}",
+            "collectioncolumn": f"{analyser_parent}/collectioncolumn",
+            "energydispersion": f"{analyser_parent}/energydispersion",
+            "detector": f"{analyser_parent}/detector",
+            "manipulator": f"{instrument_parent}/manipulator",
+            "calibration": f"{instrument_parent}/calibration",
+            "sample": f"{region_parent}/sample",
+            "data": f"{region_parent}/data",
+            "region": f"{region_parent}",
+        }
+
+        for grouping, spectrum_keys in key_map.items():
+            root = path_map[str(grouping)]
+            for spectrum_key in spectrum_keys:
+                try:
+                    mpes_key = spectrum_key
+                    self._xps_dict[f"{root}/{mpes_key}"] = spectrum[spectrum_key]
+                except KeyError:
+                    pass
+
+        if self.parser.export_settings["Transmission Function"]:
+            path = f"{path_map['collectioncolumn']}/transmission_function"
+            self._xps_dict[path] = spectrum["data"]["transmission_function"]
+            self._xps_dict[f"{path}/units"] = spectrum["tf_units"]
+
+        # Create keys for writing to data and detector
+        entry = construct_entry_name(region_parent)
+        scan_key = construct_data_key(spectrum)
+        detector_data_key_child = construct_detector_data_key(spectrum)
+        detector_data_key = f'{path_map["detector"]}/{detector_data_key_child}/counts'
+
+        x_units = spectrum["x_units"]
+        energy = np.array(spectrum["data"]["x"])
+        intensity = np.array(spectrum["data"]["y"])
```

Comment:
Probably you check for available data and handle the case if no data their.

Reply:
In the `XyProdigyParser` parser below, this is already being handled, i.e., if the spectrum has no data, it is not added to the raw_data of `XyProdigyParser`. Therefore, in the line that you are reviewing, `spectrum["data"]["y"] will always contain data (because otherwise this spectrum wouldn't exist).

So, there is no reason for having an additional check here.